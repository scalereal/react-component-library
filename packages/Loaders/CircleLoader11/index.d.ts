import React from 'react';
import { WithStyle } from '../../utils';
import { Props } from '../types';
declare const CircleLoader11: React.FC<Props> & WithStyle;
export default CircleLoader11;
