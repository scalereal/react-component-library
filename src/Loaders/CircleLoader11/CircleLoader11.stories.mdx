import CircleLoader11 from '.';
import { SizeMap } from '../types';
import { Preview, Story, Meta, Props, ArgsTable, Canvas } from '@storybook/addon-docs/blocks';
import { withKnobs, color, boolean, select } from '@storybook/addon-knobs';

<Meta
    title="Components/Loaders/CircleLoader11"
    component={CircleLoader11}
    argTypes={{
        size: {
            control: {
                type: 'select',
                options: Object.values(SizeMap)
            }
        },
        Color: {
            control: 'color'
        },
        display: {
            control: {
                type: 'boolean'
            }
        },
        ref: {
            table: {
                disable: true
            }
        },
        theme: {
            table: {
                disable: true
            }
        },
        as: {
            table: {
                disable: true
            }
        }
    }}
/>

# Loaders

Displays a Loader.

export const Template = args => (
    <div>
        <CircleLoader11 {...args} />
    </div>
);

<Preview>
    <Story name="CircleLoader11" args={{ size: 'S' }}>
        {Template.bind({})}
    </Story>
</Preview>

### Color Variations

You can either change the theme or pass the colors as props.

<Preview withToolbar>
    <div style={{ display: 'flex', justifyContent: 'space-around', flexWrap: 'wrap' }}>
        <CircleLoader11 size="S" Color="green" />
        <CircleLoader11 size="S" Color="red" />
    </div>
</Preview>

### Sizes

There are Six sizs available `XXS` | `XS` | `S` | `M` | `L` | `XL`.

<Preview withToolbar>
    <div style={{ display: 'flex', justifyContent: 'space-around', flexDirection: 'column' }}>
        <CircleLoader11 size="XXS" />
        <CircleLoader11 size="XS" />
        <CircleLoader11 size="S" />
        <CircleLoader11 size="M" />
        <CircleLoader11 size="L" />
        <CircleLoader11 size="XL" />
    </div>
</Preview>

### Props

<Props story="CircleLoader11" />
