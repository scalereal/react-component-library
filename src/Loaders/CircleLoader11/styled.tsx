import styled from 'styled-components';
import { Props } from '../types';

export const Loader = styled.div<Props>`
    display: inline-block;
    width: 1em;
    height: 1em;
    color: ${pr=>pr.Color?pr.Color:'inherit'};
    vertical-align: middle;
    pointer-events: none;
    
    border-radius: 50%;
	animation-fill-mode: both;
	animation: loader-13 1.8s infinite ease-in-out;
    &:after,&:before{
        border-radius: 50%;
        animation-fill-mode: both;
        animation: loader-13 1.8s infinite ease-in-out;
    }

    /* color: currentcolor; */
    position: relative;
    transform: translateZ(0);
    animation-delay: -0.16s;
    &:before {
        right: 100%;
        animation-delay: -0.32s;
    }
    &:after {
        left: 100%;
    }
    &:before,
    &:after {
        content: '';
        display: block;
        position: absolute;
        top: 0;
        width: inherit;
        height: inherit;
    }

    @keyframes loader-13 {
        0%,
        80%,
        100% {
            box-shadow: 0 1em 0 -1em;
        }
        40% {
            box-shadow: 0 1em 0 -.2em;
        }
    }

`;

const mixin = (size:any) => {
    switch (size) {
        case "XXS":
            return `font-size: 0.8rem;
            ${Loader}{
                margin:0 15px;
                top: -1em;
            }
           `;
        case "XS":
            return `font-size: 1rem;
            ${Loader}{
                margin:0 18px;
                top: -1em;
            }
            
            `;
        case "S":
            return `font-size: 2rem;
            ${Loader}{
                margin:0 25px;
                top: -1.1em;
            }
            `;
        
        case "M":
            return `font-size: 3rem;
            ${Loader}{
                margin:0 40px;
                top: -1.1em;
            }
            `;

        case "L":
            return `font-size: 5rem;
            ${Loader}{
                margin:0 65px;
                top: -1.2em;
            }
            `;
        
        case "XL":
            return `font-size: 6rem;
            ${Loader}{
                margin:0 80px;
                top: -1.2em;
            }
            `;
  
      default:
        return `font-size: 1rem;
        ${Loader}{
            margin:0 25px;
        }
       `;
     }
 };

export const LoaderWrapper = styled.div<Props>`
    ${pr => pr.disabled === true?'pointer-events: none;opacity: 0.4;':''}
    ${pr=>  mixin(pr.size)}
    display: ${pr=>pr.display?'inline-block':'none'};
`;
