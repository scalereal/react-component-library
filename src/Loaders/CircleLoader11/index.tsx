import React from 'react';
import { WithStyle } from '../../utils';
import { Props } from '../types';
import { Loader, LoaderWrapper } from './styled';

const CircleLoader11:React.FC<Props> & WithStyle = React.memo(React.forwardRef( (props, ref) => {
    const {size} = props;

    return (
        <LoaderWrapper ref={ref} size={size} {...props}>
            <Loader {...props}/>
        </LoaderWrapper>
    );
    
}));

CircleLoader11.displayName="CircleLoader11";
CircleLoader11.defaultProps = {
    size:'S',
    display: true
};

export default CircleLoader11;
