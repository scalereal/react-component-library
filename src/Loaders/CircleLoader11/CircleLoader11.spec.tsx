import { render } from '@testing-library/react';
import * as React from 'react';
import CircleLoader11 from '.';

describe('CircleLoader11 Loader', () => {
    it('should render with default theme', () => {
        const { container } = render(<CircleLoader11/>);
        expect(container).toMatchSnapshot();
    });
    it('should render with all the props given', () => {
        const { container } = render(
            <CircleLoader11
                size="M"
                Color='#126AFA'
            />
        );
        expect(container).toMatchSnapshot();
    });

    it('should render image avatar properly', () => {
        const { container } = render(
            <CircleLoader11 size="L" />
        );
        expect(container).toMatchSnapshot();
    });

});