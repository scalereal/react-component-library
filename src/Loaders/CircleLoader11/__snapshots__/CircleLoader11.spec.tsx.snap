// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircleLoader11 Loader should render image avatar properly 1`] = `
<div>
  <div
    class="sc-gsTCUz ecHdod"
  >
    <div
      class="sc-bdfBwQ jYKpXM"
    />
  </div>
</div>
`;

exports[`CircleLoader11 Loader should render with all the props given 1`] = `
<div>
  <div
    class="sc-gsTCUz jQTQoX"
  >
    <div
      class="sc-bdfBwQ hRrWGF"
    />
  </div>
</div>
`;

exports[`CircleLoader11 Loader should render with default theme 1`] = `
<div>
  <div
    class="sc-gsTCUz cKEXNV"
  >
    <div
      class="sc-bdfBwQ jYKpXM"
    />
  </div>
</div>
`;
